### DISCUSSION TITLE
mobx store can not work with react suspense

### DISCUSSION THREAD

**Discussion by GuichiZhao** [mobxjs/mobx#3461] [Q&A]
Here is my code:

    import { makeAutoObservable } from "mobx";
    
    class EventStore{
      constructor(){
        makeAutoObservable(this)
      }
      get events() {
        if (!this._events) {
          throw when(() => !this._events);
        }
        return this._events;
      }
    }
It is pretty valid code and works fine with react suspense, however, mobx give a annoying error is the console

> react_devtools_backend.js:4026 [mobx] Encountered an uncaught
> exception that was thrown by a reaction or observer component

Is there anything I can do?

**Comment by mweststrate** [comment]
Yes, create a reproduction :) It is unclear what the exception is, and whether it is caused by MobX, or how it relates to React.
